packages/ag-charts-community/src/chart/mapping/themes.ts

```diff
@@ -42,12 +47,72 @@ export const themes: ThemeMap = {
     ...lightThemes,
 };
 
-export function getChartTheme(value?: string | ChartTheme | AgChartTheme): ChartTheme {
+type Unvalidate<T> = { [K in keyof T]?: unknown };
+function validateChartThemeObject(unknownObject: object | null): AgChartTheme | undefined {
+    if (unknownObject === null) {
+        return undefined;
+    }
+
+    let valid = true;
+    const { baseTheme, palette, overrides } = unknownObject as Unvalidate<AgChartTheme>;
+
+    if (baseTheme !== undefined && typeof baseTheme !== 'string' && typeof baseTheme !== 'object') {
+        Logger.warn(`invalid theme.baseTheme type ${typeof baseTheme}, expected (string | object).`);
+        valid = false;
+    }
+    if (overrides !== undefined && typeof overrides !== 'object') {
+        Logger.warn(`invalid theme.overrides type ${typeof overrides}, expected object.`);
+        valid = false;
+    }
+    if (typeof palette === 'object') {
+        if (palette !== null) {
```

Comment:
Nit: The `typeof palette === 'object'` check makes this guard redundant?

Reply:
You'd think so, but my `tsserver` tells me that the type of `palette` is `object | null` after the first guard.

This might be okay, but I don't know how the `const { fills, strokes } = palette` deconstruction will behave if the palette is `null`.